circuits/kimchi/src/expr.rs

```diff
@@ -1947,3 +1951,308 @@ impl<F: fmt::Display + Clone> fmt::Display for Expr<F> {
+                ),
+                PolishToken::Literal(x) => CamlPolishToken::Literal(x.into()),
+                PolishToken::Cell(x) => CamlPolishToken::Cell(x.into()),
+                PolishToken::Dup => CamlPolishToken::Dup,
+                PolishToken::Pow(x) => CamlPolishToken::Pow(x.try_into().expect("isize -> usize")),
+                PolishToken::Add => CamlPolishToken::Add,
+                PolishToken::Mul => CamlPolishToken::Mul,
+                PolishToken::Sub => CamlPolishToken::Sub,
+                PolishToken::VanishesOnLast4Rows => CamlPolishToken::VanishesOnLast4Rows,
+                PolishToken::UnnormalizedLagrangeBasis(x) => {
+                    CamlPolishToken::UnnormalizedLagrangeBasis(
+                        x.try_into().expect("isize -> usize"),
+                    )
+                }
+                PolishToken::Store => CamlPolishToken::Store,
+                PolishToken::Load(x) => {
+                    CamlPolishToken::Load(x.try_into().expect("isize -> usize"))
+                }
+            }
+        }
+    }
+
+    impl<F, CamlF> From<&PolishToken<F>> for CamlPolishToken<CamlF>
+    where
+        CamlF: From<F>,
+        F: Copy,
+    {
+        fn from(c: &PolishToken<F>) -> CamlPolishToken<CamlF> {
+            match c {
+                PolishToken::Alpha => CamlPolishToken::Alpha,
+                PolishToken::Beta => CamlPolishToken::Beta,
+                PolishToken::Gamma => CamlPolishToken::Gamma,
+                PolishToken::JointCombiner => CamlPolishToken::JointCombiner,
+                PolishToken::EndoCoefficient => CamlPolishToken::EndoCoefficient,
+                PolishToken::Mds { row, col } => CamlPolishToken::Mds(
+                    (*row).try_into().expect("isize -> usize"),
+                    (*col).try_into().expect("isize -> usize"),
+                ),
+                PolishToken::Literal(x) => CamlPolishToken::Literal((*x).into()),
+                PolishToken::Cell(x) => CamlPolishToken::Cell(x.into()),
+                PolishToken::Dup => CamlPolishToken::Dup,
+                PolishToken::Pow(x) => {
+                    CamlPolishToken::Pow((*x).try_into().expect("isize -> usize"))
+                }
+                PolishToken::Add => CamlPolishToken::Add,
+                PolishToken::Mul => CamlPolishToken::Mul,
+                PolishToken::Sub => CamlPolishToken::Sub,
+                PolishToken::VanishesOnLast4Rows => CamlPolishToken::VanishesOnLast4Rows,
+                PolishToken::UnnormalizedLagrangeBasis(x) => {
+                    CamlPolishToken::UnnormalizedLagrangeBasis(
+                        (*x).try_into().expect("isize -> usize"),
+                    )
+                }
+                PolishToken::Store => CamlPolishToken::Store,
+                PolishToken::Load(x) => {
+                    CamlPolishToken::Load((*x).try_into().expect("isize -> usize"))
+                }
+            }
+        }
+    }
+}
```

Comment:
so, do we really need to expose these types to the OCaml side? What do we really need on the OCaml side?